pallets/oracle/src/lib.rs

```diff
@@ -57,10 +59,16 @@ pub mod types;
 pub mod dia;
 
 pub mod oracle_api;
+
 pub use crate::oracle_api::*;
+
 #[cfg(feature = "testing-utils")]
 pub mod oracle_mock;
 
+// We assume this value to be the decimals of our base currency (USD). It doesn't matter too much as
+// long as it's consistent.
+const USD_DECIMALS: u32 = 12;
```

Comment:
It's not even linked to how the prices are stored in the DIA oracle. This constant will just determine what we consider 1 unit of USD when calling either the `currency_to_usd()` or `usd_to_currency` function. Since going forward, these functions are only used by the exposed RPC API (because the `convert()` function itself doesn't call them anymore), the `USD_DECIMALS` are only relevant to interpret the result of the RPC call. We can set it to whatever we want without impacting any functionality. 

For example, assuming we keep the `USD_DECIMALS` as `12`, here we would interpret that 1 XLM is worth $ `0.000,000,116,964` (this was taken from the current setup where we have this 6 decimal shifting problem solved by the [other](https://github.com/pendulum-chain/pendulum/pull/413) ticket).
<img width="1702" alt="image" src="https://github.com/pendulum-chain/spacewalk/assets/6690623/0c835516-e50b-40b4-872e-71f550a447db">

Reply:
Understood. Thanks for the explanation. So although it is only relevant for these two RPC functions it would be nice to set it up so that such that the output here reflects the true usd value of the token. In this case when the other fix comes online, `12` decimals will not be enough for this I imagine right? Since the price is divided by 12 yet here we consider one unit of USD = 1, and not 1/10**12.

I tried to follow the calculation but got a little lost, so not sure if this is correct.

Reply:
I'll try to explain again how this relates to the change in https://github.com/pendulum-chain/pendulum/pull/413.
By changing the FixedPoint creation to `from_rational()` instead of `from_inner()` in https://github.com/pendulum-chain/pendulum/pull/413, the exchange rates will be represented how they should be internally. So let's say, our DIA pallet has an on-chain price of `123,000,000,000` ($0.123) stored for XLM. Before the change (with `from_inner()`) the fixed point number would be `0.000000123`. With the change to `from_rational()` it will be `0.123`. 

Now, let's look at a call of `currencyToUsd()` converting from XLM to USD, assuming that we are using the 'proper' fixed point number of `0.123`. If we set `USD_DECIMALS` to `12`, the result of `currency_to_usd()` will be `123,000,000,000`. If we set `USD_DECIMALS` to `18`, the result will be `123,000,000,000,000,000`. If we set `USD_DECIMALS` to `9`, the result will be `123,000,000`. What I'm trying to illustrate is that the actual value of this parameter does not matter for the calculation but it matters for how we interpret the result.